src/IO/ReadBufferFromS3.h

```diff
@@ -49,6 +49,7 @@ class ReadBufferFromS3 : public ReadBufferFromFileBase
         const String & version_id_,
         const S3Settings::RequestSettings & request_settings_,
         const ReadSettings & settings_,
+        String buffer_creator_name,
```

Comment:
Maybe `std::string_view` or `const char *`, because this `creator_name` is always a constant.